src/main/java/org/hiero/sketch/table/ObjectArrayColumn.java

```diff
@@ -0,0 +1,129 @@
+package org.hiero.sketch.table;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.hiero.sketch.table.api.IStringConverter;
+import org.hiero.sketch.table.api.IndexComparator;
+
+import java.time.Duration;
+import java.util.Date;
+
+/*
+ * Column of objects of any type; only for moving data around. Size of column expected to be small.
+ */
+public final class ObjectArrayColumn extends BaseArrayColumn {
+    @NonNull private final Object[] data;
+
+    public ObjectArrayColumn(@NonNull final ColumnDescription description, final int size) {
+        super(description, size);
+        this.data = new Object[size];
+    }
+
+    public ObjectArrayColumn(@NonNull final ColumnDescription description,
+                             @NonNull final Object[] data) {
+        super(description, data.length);
+        this.data = data;
+    }
+
+    @Override
+    public int sizeInRows() { return this.data.length; }
+
+    @Override
+    public double asDouble(final int rowIndex, @NonNull final IStringConverter converter) {
+        switch (ObjectArrayColumn.this.description.kind) {
+            case String:
+                return converter.asDouble(this.getString(rowIndex));
+            case Date:
+                return converter.asDouble(this.getDate(rowIndex).toString());
+            case Int:
+                return converter.asDouble(String.valueOf(this.getInt(rowIndex)));
+            case Json:
+                return converter.asDouble(this.getString(rowIndex));
+            case Double:
+                return this.getDouble(rowIndex);
+            case Duration:
+                return converter.asDouble(this.getDuration(rowIndex).toString());
+            default:
+                throw new RuntimeException("Unexpected data type");
+        }
+    }
+
+    @Override
+    public String asString(final int rowIndex) {
+        return this.data[rowIndex].toString();
+    }
+
+    @Override
```

Comment:
Can you add a method getObjectComparator which returns an IComparator<Object>?
You can use that in comparing across two columns.